pkg/provider/vault/vault.go

```diff
@@ -1271,6 +1279,33 @@ func setIamAuthToken(ctx context.Context, v *client, jwtProvider util.JwtProvide
 	return false, nil
 }
 
+func setGcpAuthToken(ctx context.Context, v *client) (bool, error) {
+	gcpAuth := v.store.Auth.Gcp
+	var ts oauth2.TokenSource
+	var err error
+	if gcpAuth != nil {
+		switch {
+		case gcpAuth.SecretRef != nil:
+			ts, err = gcpauth.ServiceAccountCredentials(ctx, v.kube, v.namespace, gcpAuth.SecretRef.SecretAccessKey)
+		case gcpAuth.WorkloadIdenity != nil:
+			ts, err = gcpauth.WorkloadIdentityCredentials(ctx, v.kube, v.namespace, *gcpAuth.WorkloadIdenity)
+		// Default to Pod identity if no auth method is specified.
```

Comment:
I don't see any restriction on these two being defined at the same time, right? Which means, WorkloadIdentity would just be ignored since the switch just runs SecretRef and that's it.

We probably need some kind of warning in that case.